pulsar-broker/src/main/java/org/apache/pulsar/broker/offload/impl/ManagedLedgerOffloader.java

```diff
@@ -96,23 +122,66 @@ public static S3ManagedLedgerOffloader create(ServiceConfiguration conf,
             throw new PulsarServerException("s3ManagedLedgerOffloadMaxBlockSizeInBytes cannot be less than 5MB");
         }
 
-        AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
+        return new ManagedLedgerOffloader(driver, bucket, scheduler, maxBlockSize, readBufferSize, endpoint, region);
+    }
+
+    // build context for jclouds BlobStoreContext
+    ManagedLedgerOffloader(String driver, String container, OrderedScheduler scheduler,
+                           int maxBlockSize, int readBufferSize, String endpoint, String region) {
+        this.scheduler = scheduler;
+        this.readBufferSize = readBufferSize;
+
+        this.bucket = container;
+        this.maxBlockSize = maxBlockSize;
+
+        Properties overrides = new Properties();
+        // This property controls the number of parts being uploaded in parallel.
+        overrides.setProperty("jclouds.mpu.parallel.degree", "1");
+        overrides.setProperty("jclouds.mpu.parts.size", Integer.toString(maxBlockSize));
+        overrides.setProperty(Constants.PROPERTY_SO_TIMEOUT, "25000");
+        overrides.setProperty(Constants.PROPERTY_MAX_RETRIES, Integer.toString(100));
+
+        ContextBuilder contextBuilder = ContextBuilder.newBuilder(driver);
+
+        AWSCredentials credentials = null;
+        try {
+            DefaultAWSCredentialsProviderChain creds = DefaultAWSCredentialsProviderChain.getInstance();
+            credentials = creds.getCredentials();
+        } catch (Exception e) {
+            log.error("Exception when get credentials for s3 ", e);
+        }
+
+        String id = "accesskey";
```

Comment:
default these to ""

Reply:
Thanks. will keep this default value. empty string will meet error in jclouds.